### PR TITLE
Hover tooltips

### PR DESCRIPTION
**Pull request by hassanbazzi**

### CODE REVIEW

File: dist/utils.js [hunk 1]
```diff
@@ -397,6 +422,18 @@ angular.module('ts.utils').directive('focusOn', function ($window, focusOnConfig
...
+'use strict';
+
+(function (module) {
+  try {
+    module = angular.module('ts.utils');
+  } catch (e) {
+    module = angular.module('ts.utils', []);
+  }
+  module.run(['$templateCache', function ($templateCache) {
+    $templateCache.put('templates/tsTooltip.html', '<div class="ts-tooltip-container {{::tsTooltipClass}}">\n' + '  <div class="arrow-box-container">\n' + '    <div id="tooltipMain" class="ts-tooltip-main">\n' + '      {{tsTooltip}}\n' + '    </div>\n' + '  </div>\n' + '</div>\n' + '');
```
Comment (shusain) [inline dist/utils.js:434]: Did you miss committing the source file here? was expecting to see the source template updated with the tsTooltipClass but only see it here.

File: src/tooltip.js [hunk 2]
```diff
@@ -133,21 +137,44 @@ angular.module('ts.utils')
...
-        function makeVisible(){
-          if(!isVisible) {
-            document.body.insertBefore(tooltipContainer[0],document.body.childNodes[0]);
+        function makeVisible() {
+          if (!isVisible) {
+            document.body.insertBefore(tooltipContainer[0], document.body.childNodes[0]);
             positionTooltip();
             isVisible = true;
+            if($scope.tsTooltipContentHover) {
+              angular.element(tooltipContainer).bind('mouseleave', function(){
```
Comment (shusain) [inline src/tooltip.js:146]: Do these event listeners get unbound automatically when the element is removed from the DOM?  Concern here is stacking up event listeners every time the makeVisible() is called.

File: dist/utils.js (hunk 1 above)
Comment (hassanbazzi) [inline dist/utils.js:434]: what do you mean? I messed it up last time yea

File: src/tooltip.js (hunk 2 above)
Comment (hassanbazzi) [inline src/tooltip.js:146]: yea they do. When you remove something from the dom, the listeners are destroyed as well

**Pull request merged**